Core/Src/main.c

```diff
@@ -87,20 +99,46 @@ int main(void)
   /* Initialize all configured peripherals */
   MX_GPIO_Init();
   MX_USART2_UART_Init();
+  MX_SPI1_Init();
+  MX_TIM1_Init();
+
   /* USER CODE BEGIN 2 */
 
+
+  HAL_TIM_PWM_Start(&htim, TIM_CHANNEL_1);
+  /*send clock signal */
+
   /* USER CODE END 2 */
 
   /* Infinite loop */
   /* USER CODE BEGIN WHILE */
   while (1)
-  {
+
+	  HAL_GPIO_WritePin(GPIOB, GPIO_PIN_8, 0);
+
+  	  HAL_SPI_TransmitReceive(hspi, txData, rxData, 3, 1000);
+
+
+  	  HAL_GPIO_WritePin(GPIOB, GPIO_PIN_8, 1);
+
+  	  //rxData[1] contains 2 useful bits, txData[2] contains 8 useful, lower order bits
+
+  	  adcValue = ((rxData[1] & 0x03) << 8) | rxData[2];
+
+
+  	  counts = TIM_1MS + (adcValue/ADC_MAX)*(TIM_2MS - TIM_1MS);
+
+  	  _HAL_TIM_SET_COMPARE(&htim, TIM_CHANNEL_1, counts);
+
+
+	  TIM1->CCR1 = 20;
```

Comment:
Everything looks good, may I ask what this line is for? Don't Hal Tim Set Compare sets the TIM Capture Compare Register value on runtime already?

Reply:
I was watching a tutorial and I saw that this line was included. I didn't know what its function was at the time, now that you mention it I don't know why it's there either. I must have just forgotten to remove it after I wrote the rest of the code